### ISSUE TITLE
Crash at FlxSound line 172 when playing sounds

### ISSUE THREAD

**Issue by Gama11** [FlixelCommunity/flixel#138]
Got a copy of the current build and tried to use it in a project. However, whenever playing a sound, there's a null error crash at line 172 in the `destroy`-function of `FlxSound`:

``` actionscript
_sound.removeEventListener(Event.ID3, gotID3);
```

Played around with it a bit. It doesn't matter whether the sound is played via `FlxG.play` or if you instantiate a new `FlxSound` without even playing it, it'll always crash. It even crashes when changing the game's volume as there's a built-in soundeffect for that.

It seems like if a sound is created via `loadEmbedded`, which is the most likely scenario, no ID3 event listener is created like in `loadStream`:

``` actionscript
_sound.addEventListener(Event.ID3, gotID3);
```

There's a note in line 270:

``` actionscript
//NOTE: can't pull ID3 info from embedded sound currently
```

As no event listener exists, it is `null` and the attempt to remove it will cause a crash. Could be fixed by adding a `private var isStreamed:Boolean = false` which is set to true in `loadStream`, then checking whether the sound has been loaded from a stream or not in `destroy`:

``` actionscript
if (isStreamed) _sound.removeEventListener(Event.ID3, gotID3);
```

**Comment by IQAndreas** [comment]
Actually, in that case, I believe `_sound` is `null` by the time `destroy()` is called, resulting in the error. Though, looking through the code, that issue no longer occurs as of one of my pull requests that modifies FlxSound.

Give this SWC a try and let me know if the error still occurs: https://github.com/FlixelCommunity/flixel-release/tree/bin-dev

And, for reference, removing an event listener that does not exist does not produce any errors. It simply does nothing and keeps going, so no need to worry there.

**Comment by Gama11** [comment]
> Actually, in that case, I believe _sound is null by the time destroy() is called, resulting in the error. Though, looking through the code, that issue no longer occurs as of one of my pull requests that modifies FlxSound.

Seems like you're right. I just confirmed that with `FlxG.log(_sound)`. A simple `null` check would be enough to prevent the error.

> Give this SWC a try and let me know if the error still occurs: https://github.com/FlixelCommunity/flixel-release/tree/bin-dev

Nope, didn't occur when compiling a swf from that swc. Why is it only fixed there and not for the main repo? Has that pull request you're talking about not been merged yet?

> And, for reference, removing an event listener that does not exist does not produce any errors. It simply does nothing and keeps going, so no need to worry there.

That's good to know, thanks. I swear I tried adding a simple `if (_sound != null)` earlier, but it didn't work...

**Comment by IQAndreas** [comment]
> Why is it only fixed there and not for the main repo? Has that pull request you're talking about not been merged yet?

Exactly, it's still open (see https://github.com/FlixelCommunity/flixel/pull/114).

But thanks for pointing this out. In the current pull request, I realized that I forgot to destroy the Sound in the `destroy()` function, so I will have to add that.

**Comment by IQAndreas** [comment]
Resolved in 7e5f1e0504151e8e9b306a444c5eaa6e10817472
